File: Backend/src/main/java/com/example/backend/entity/RecipeEntity.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "recipes")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RecipeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 150)
    private String title;

    @Column(length = 500)
    private String description;

    @Column(length = 50)
    private String author;

    @Column(name = "preview_image_url", length = 200)
    private String previewImageUrl;

    @Column(name = "cooking_time_minutes", nullable = false)
    private Integer cookingTimeMinutes;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "likes_count")
    private Integer likesCount = 0;

    @ManyToMany
    @JoinTable(
            name = "recipe_category_cross",
            joinColumns = @JoinColumn(name = "recipe_id"),
            inverseJoinColumns = @JoinColumn(name = "category_id")
    )
    private List<CategoryEntity> categories = new ArrayList<>();

    @OneToMany(mappedBy = "recipe", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<RecipeGroceryCrossEntity> recipeGroceries = new ArrayList<>();

    @OneToOne(mappedBy = "recipe", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private RecipeContentEntity content;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: MyApplication/data/src/main/java/com/example/data/Recipes/model/RecipeEntity.kt
package com.example.data.Recipes.model

import com.example.domain.Recipes.model.Recipe
import com.example.domain.Recipes.model.RecipeIngredient
import java.time.LocalDateTime

data class RecipeEntity(
    val id: Int,
    val title: String,
    val description: String?,
    val author: String,
    val previewImageUrl: String?,
    val cookingTimeMinutes: Int,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val likesCount: Int = 0
) {
    fun toDomain(
        categories: List<CategoryEntity> = emptyList(),
        ingredients: List<RecipeIngredient> = emptyList(),
        isFavorite: Boolean = false,
        isLiked: Boolean = false
    ): Recipe {
        return Recipe(
            id = id,
            title = title,
            description = description,
            categories = categories.map { it.toDomain() },
            ingredients = ingredients,
            author = author,
            previewImageUrl = previewImageUrl,
            cookingTimeMinutes = cookingTimeMinutes,
            createdAt = createdAt,
            updatedAt = updatedAt,
            likesCount = likesCount,
            isFavorite = isFavorite,
            isLiked = isLiked
        )
    }
}